src/main/java/com/microsoft/util/Utils.java

```diff
@@ -24,4 +52,157 @@ public static boolean isPublic(Element e) {
+        ElementKind kind = encl.getKind();
+        if (kind == ElementKind.PACKAGE)
+            return null;
+        while (!(kind.isClass() || kind.isInterface())) {
+            encl = encl.getEnclosingElement();
+            kind = encl.getKind();
+        }
+        return (TypeElement) encl;
+    }
+
+    /**
+     * Follow the same behavior with Standard doclet.
+     * Return the ExecutableElement being overridden by given method,
+     * when a method in a class overrides a method in a superclass.
+     * For following cases, they will be marked as "Specified by":
+     * <ul>
+     * <li>a method in an interface overrides a method in a superinterface</li>
+     * <li>a method in a class implements a method in an interface</li>
+     * </ul>
+     * todo add "Specified by" to yaml.
+     */
+    public ExecutableElement overriddenMethod(ExecutableElement method) {
+        if (isStatic(method)) {
+            return null;
+        }
+        final TypeElement origin = getEnclosingTypeElement(method);
+        for (TypeMirror t = getSuperType(origin);
+             t.getKind() == DECLARED;
+             t = getSuperType(asTypeElement(t))) {
+            TypeElement te = asTypeElement(t);
+            if (te == null) {
+                return null;
+            }
+
+            for (Element e : getMembers(te, ElementKind.METHOD)) {
+                ExecutableElement ee = (ExecutableElement) e;
+                if (elementUtils.overrides(method, ee, origin)
+                ) {
+                    return ee;
+                }
+            }
+            if (t.equals(getObjectType().asType()))
+                return null;
+        }
+        return null;
+    }
+
+    public TypeMirror getSuperType(TypeElement te) {
+        TypeMirror t = te.getSuperclass();
+        return getType(t);
+    }
+
+    private TypeMirror getType(TypeMirror t) {
+        return (isNoType(t)) ? getObjectType().asType() : t;
+    }
+
+    public Optional<DocCommentTree> getDocCommentTree(Element element) {
+        return Optional.ofNullable(docletEnvironment.getDocTrees().getDocCommentTree(element));
+    }
+
+    protected List<? extends DocTree> getFullBody(Element element) {
+        return getDocCommentTree(element)
+                .map(DocCommentTree::getFullBody)
+                .orElse(Collections.emptyList());
+    }
+
+    public static List<? extends DocTree> filteredList(List<? extends DocTree> dlist, DocTree.Kind... select) {
+        List<DocTree> list = new ArrayList<>();
+        if (select == null)
+            return dlist;
+        for (DocTree dt : dlist) {
+            if (dt.getKind() != ERRONEOUS) {
+                for (DocTree.Kind kind : select) {
+                    if (dt.getKind() == kind) {
+                        list.add(dt);
+                    }
+                }
+            }
+        }
+        return list;
+    }
+
+    private List<? extends DocTree> getBlockTags0(Element element, DocTree.Kind... kinds) {

```

Comment:
What do you mean by 0?

Reply:
This is re-use of Standard doclet code, I have the same question of what "0" stands for. But I did not find an answer yet, so keep the same name for now.